base/error.jl

```diff
@@ -91,28 +91,34 @@ function catch_backtrace()
     return _reformat_bt(bt[], bt2[])
 end
 
+struct ExceptionStack <: AbstractArray{Any,1}
```

Comment:
This type doesn't seem necessary to me. Since showing errors goes through `display_error` anyway, having a type with a `show` method is a bit redundant (and does have some duplicated code here).

Reply:
Not that I love `ExceptionStack` as a type, it seems annoying and superfluous on the surface, and it's one of the main reasons I wanted some more eyes on this.

What I'd really like are suggestions about how to replace or improve it, while also having a way to pass these to logging macros and have them robustly recognized by the logger backend.

Reply:
It's true that there's a little duplication of code between `show` and `display_error`. Previously I tried making `display_error` dispatch to `show` and and adding some `IOContext` handling to turn on the simplifications we currently use in `display_error`. That turned out somewhat complex though so I punted in the current iteration and left it with to concrete implementations.

I could give it another go.